src/Microsoft.Identity.Web.TokenCache/Distributed/MsalDistributedTokenCacheAdapter.cs

```diff
@@ -263,14 +263,29 @@ protected override async Task WriteCacheBytesAsync(string cacheKey, byte[] bytes
                 SlidingExpiration = _distributedCacheOptions.SlidingExpiration,
             };
 
-            await L2OperationWithRetryOnFailureAsync(
-            write,
-            (cacheKey) => _distributedCache.SetAsync(
-                cacheKey,
-                bytes,
-                distributedCacheEntryOptions,
-                cacheSerializerHints?.CancellationToken ?? CancellationToken.None),
-            cacheKey).Measure().ConfigureAwait(false);
+            if (_distributedCacheOptions.DisableL1Cache)
+            {
+                await L2OperationWithRetryOnFailureAsync(
+                    write,
+                    (cacheKey) => _distributedCache.SetAsync(
+                        cacheKey,
+                        bytes,
+                        distributedCacheEntryOptions,
+                        cacheSerializerHints?.CancellationToken ?? CancellationToken.None),
+                    cacheKey).Measure().ConfigureAwait(false);
+            }
+            else
+            {
+               _ = Task.Run(async () => await
+                L2OperationWithRetryOnFailureAsync(
+                write,
+                (cacheKey) => _distributedCache.SetAsync(
+                    cacheKey,
+                    bytes,
+                    distributedCacheEntryOptions,
+                    cacheSerializerHints?.CancellationToken ?? CancellationToken.None),
+                cacheKey).Measure().ConfigureAwait(false));
```

Comment:
Why is Measure being called twice? It seems to be called once here and once in `L2OperationWithRetryOnFailureAsync`. Does it still work in fire and forget mode?

Reply:
yes it still works based on my testing, compared with before.